libknet/crypto.c

```diff
@@ -138,8 +140,10 @@ int crypto_init(
 	 * crypto_modules_cmds.ops->fini is not invoked on error.
 	 */
 	knet_h->crypto_instance->model = model;
-	if (crypto_modules_cmds[knet_h->crypto_instance->model].ops->init(knet_h, knet_handle_crypto_cfg))
+	if (crypto_modules_cmds[knet_h->crypto_instance->model].ops->init(knet_h, knet_handle_crypto_cfg)) {
+		savederrno = EPIPE;
```

Comment:
In theory ops->init should return the correct errno, but clearly that's not true everywhere. Instead of enforcing one from the main cryto module, we should fix the modules themselves to return the right error.